packages/firebase_storage/firebase_storage_web/lib/src/interop/storage.dart

```diff
@@ -488,17 +473,30 @@ abstract class _SettableMetadataBase<
   }
 
   /// Additional user-defined custom metadata.
-  Map<String, String> get customMetadata =>
-      (dartify(jsObject.customMetadata) as Map)?.cast<String, String>();
+  Map<String, String> get customMetadata {
+    Map<String, String>? metadata = dartify(jsObject.customMetadata);
+    if (metadata != null) {
+      return metadata;
+    } else {
+      return {};
+    }
+  }
```

Comment:
I updated all this logic because this: `dartify(jsObject.customMetadata) as Map` can easily be `null` which didn't cause a static analysis error when casting, but if I ran it this code in dartpad, it did:
```dart
var map = null;
  
  var thing = (map as Map).cast<String, String>();
  
  thing['stuff'] = "foo";
```

I also returned empty Map literal. If make the getter type nullable (`Map<String, String>? get customMetadata`), we also have to make the setter param nullable (` set customMetadata(Map<String, String>? m)`). Didn't seem right, happy to change though.